packages/sources/binance/test/integration/adapter.test.ts

```diff
@@ -76,3 +92,85 @@ describe('execute', () => {
     })
   })
 })
+
+describe('websocket', () => {
+  let mockedWsServer: InstanceType<typeof MockWsServer>
+  let server: http.Server
+  let req: SuperTest<Test>
+
+  let oldEnv: NodeJS.ProcessEnv
+  beforeAll(async () => {
+    if (!process.env.RECORD) {
+      mockedWsServer = mockWebSocketServer('wss://stream.binance.com:9443/ws')
+      mockWebSocketProvider(WebSocketClassProvider)
+    }
+
+    oldEnv = JSON.parse(JSON.stringify(process.env))
+    process.env.WS_ENABLED = 'true'
+    process.env.WS_SUBSCRIPTION_TTL = '3000'
+
+    server = await startServer()
+    req = request(`localhost:${(server.address() as AddressInfo).port}`)
+  })
+
+  afterAll((done) => {
+    process.env = oldEnv
+    nock.restore()
+    nock.cleanAll()
+    nock.enableNetConnect()
+    server.close(done)
+  })
+
+  describe('WS crypto api', () => {
+    const jobID = '1'
+
+    it('should return success', async () => {
+      const data: AdapterRequest = {
+        id: jobID,
+        data: {
+          from: 'ETH',
+          to: 'BTC',
+        },
+      }
+
+      let flowFulfilled: Promise<boolean>
+      if (!process.env.RECORD) {
+        mockRateResponseSuccess() // For the first response
+
+        flowFulfilled = mockWebSocketFlow(mockedWsServer, [
+          mockSubscribeResponse,
+          mockUnsubscribeResponse,
+        ])
+      }
+
+      const makeRequest = () =>
+        req
+          .post('/')
+          .send(data)
+          .set('Accept', '*/*')
+          .set('Content-Type', 'application/json')
+          .expect('Content-Type', /json/)
+          .expect(200)
+
+      // We don't care about the first response, coming from http request
+      // This first request will start both batch warmer & websocket
+      // TODO: Validate just in case?
```

Comment:
@alejoberardino Do we need this TODO here?

Reply:
yeah i think this should be removed